src/Knp/Rad/ResourceResolver/RoutingNormalizer.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Knp\Rad\ResourceResolver;
+
+class RoutingNormalizer
+{
+    public function normalizeDeclaration($declaration)
+    {
+        if (is_string($declaration)) {
+            return $this->normalizeString($declaration);
+        }
+
+        if (array_keys($declaration) === array_keys(array_values($declaration))) {
```

Comment:
I think I would prefer an early return, and for this part try to make it more readable (with comments or a private method?).